cgsmiles/sample.py

```diff
@@ -10,17 +11,134 @@
+    >>> polymer_graph = sampler.sample(target_weight=100000)
+
+    One can also make a blocky copolymer but steer the probability of having
+    head-to-head tail-to-tail vs head-to-tail addition. This is done by labeling
+    the bonding descriptors A-D and providing the probability that a certain
+    bonding descriptor is used to extend the polymer.
+
+    >>> cgsmiles_str = "{#PMMA=[>A]C(C)C[<B]C(=O)OC,#PS=[>C]CC[<D]c1cccc1}"
+    >>> polymer_reactivites = {">A":0.4, "<B":0.1, ">C": 0.4, "<D": 0.1}
+    >>> sampler = MoleculeSampler.from_string(cgsmiles_str)
+    >>> polymer_graph = sampler.sample(target_weight=100000)
+
+    One can also make a blocky copolymer by providing the conditional
+    probabilities. To distinguish the descriptors on PMMA vs PS we can
+    label them with a alphanumeric string.
+
+    >>> cgsmiles_str = "{#PMMA=[$A]C(C)C[$B]C(=O)OC,#PS=[$C]CC[$D]c1cccc1}"
+    >>> polymer_reactivites = {">A":0.5, "<A":0, ">B": 0.5, "<B": 0.0}
+    >>> fragment_reactivies = {'$A': {'$A': 0.,   '$C': 0.,   '$B': 0.7, '$D': 0.3},
+                               '$B': {'$A': 0.7, '$C': 0.3, '$B': 0.0, '$D': 0.0 },
+                               '$C': {'$A': 0.,   '$C': 0.,   '$B': 0.3, '$D': 0.7},
+                               '$D': {'$A': 0.3, '$C': 0.7, '$B': 0.0, '$D': 0.0 }},
+    >>> sampler = MoleculeSampler.from_string(cgsmiles_str,
+                                              polymer_reactivies=polymer_reactivies,
+                                              fragment_reactivies=fragment_reactivies)
+    >>> polymer_graph = sampler.sample(target_weight=100000)
+
+    Advanced API Examples
+    ---------------------
+    Of course the sampler is not limited to linear co-polymers. Also randomly
+    branched molecules such as bottle-brush copolymers can be generated. To
+    steer the branching ratio and branch extension as well as terminal end
+    groups the `branch_term_probs`, `terminal_fragments`, and `terminal_reactivities`
+    can be provided.
+
+    For example, To generate a bottle brush polymer that has PMA in the backbone
+    and PEG as side-chain terminated with an OH group the following CGSmiles string
+    in combination with the above mentioned probabilities can be provided.
+
+    >>> cgsmiles_str="{#PMA=[>]CC[<]C(=O)OC[>A],#PEG=[<A]COC[>A],#OH=[<A]O}"
+    >>> sampler = MoleculeSampler.from_fragment_string(cgsmiles_str,
+                                                       branch_term_probs={"PEG": 0.3, "PMA":0.0},
+                                                       terminal_fragments=['OH'],
+                                                       bonding_probabilities={'<': 0.01, '>': 0.01, '>A': 0.99, '<A':0.99},
+                                                       bond_term_probs={'<A': 1., '>A':1., '>': 0, '<': 0},
+                                                       all_atom=True)
+    >>> molecule = sampler.sample(target_weight=5008)
```

Comment:
Not sure I like the whole terminal thing. Wouldn't the following do (roughly) the same?
```
cgsmiles_str="{#PMA=[>]CC[<]C(=O)OC[>A],#PEG=[<A]COC[>A],#OH=[<T]O}"
sampler = MoleculeSampler.from_fragment_string(cgsmiles_str,
                                                       polymer_reactivities={'<': 0.01, '>': 0.01, '>A': 0.99, '<A':0.99},
                                                       fragment_reactivities={
'>': {'<': 1}, '<': {'>': 1}, '>A': {'<A': 0.7, '<T': 0.3}}  # etc.
                                                       all_atom=True)
```
The whole separation/relation between bonding probabilities and polymer reactivities is not super clear to me. What does the first allow you to do you cannot do with the second?

Reply:
CG representation where you don't have a terminal fragment.